components/omega/doc/design/MeshClass.md

```diff
@@ -0,0 +1,73 @@
+(omega-design-mesh)=
+# Mesh 
+
+## 1 Overview
+
+The mesh class will contain the YAKL arrays which describe the mesh and are used in the computation of the tendency terms in the discrete governing equations.
+
+## 2 Requirements
+
+### 2.1 Requirement: OMEGA will use the previously estabilished MPAS Mesh Spec
+
+The OMEGA mesh information should be compatiable with the [MPAS Mesh Specification](https://mpas-dev.github.io/files/documents/MPAS-MeshSpec.pdf).
```

Comment:
> My initial thought was that Decomp class instance could live inside the Mesh class

It seems like Decomp has functionality that intersects with Mesh but doesn't quite belong inside of Mesh. 
 It should be easy to have a pointer to a Decomp object in Mesh without having that object "live" in Mesh.

Reply:
@xylar - I tend to think of the decomposition as a property of a given global mesh, but I think what you say makes sense since what we really are creating here is an object for the _local_ mesh information. Perhaps the mesh constructor can take the Decomp object as an argument in order to create the associations connectivity arrays?